client/src/components/LayoutComponents/Form/index.jsx

```diff
@@ -0,0 +1,191 @@
+import React, { Fragment, Component } from 'react';
+import {
+	Checkbox,
+	FormControlLabel,
+	Grid,
+	TextField,
+	Select,
+	MenuItem,
+	InputLabel,
+	Button,
+	FormControl,
+} from '@material-ui/core';
+import axios from 'axios';
+import Alert from '@material-ui/lab/Alert';
+import Dialog from '../../CommonComponents/Dialog';
+
+class Form extends Component {
+	state = {
+		questions: [],
+		newQuestionText: '',
+		newQuestionType: '',
+		loading: true,
+		error: '',
+		typeError: '',
+		selectedQuestions: {},
+		submitDialogOpen: false,
+	};
+
+	componentDidMount = () => {
+		axios
+			.get('/dashboard/form')
+			.then(({ data }) => {
+				if (!data)
+					return this.setState({
+						loading: false,
+						error: 'There was an error loading your data',
+					});
+				this.setState({ questions: data, loading: false });
+			})
+			.catch((err) =>
+				this.setState({
+					error: 'There was an error loading your data',
+				}),
+			);
+	};
+
+	handleCheckQuestion = (question) => {
+		this.setState({
+			selectedQuestions: {
+				...this.state.selectedQuestions,
+				[question.id]: !this.state.selectedQuestions[question.id],
+			},
+		});
+	};
+
+	handleChangeNewQuestionText = (event) => {
+		let { value } = event.target;
+
+		this.setState({ newQuestionText: value });
+	};
+	handleSelectNewQuestionType = (event) => {
+		this.setState({ newQuestionType: event.target.value });
+	};
+	handleAddQuestionClick = (event) => {
+		event.preventDefault();
+		if (
+			this.state.newQuestionType === '' ||
+			this.state.newQuestionText === ''
+		) {
+			return this.setState({
+				typeError:
+					'Please insert the quesstion & choose the type of the Question!!',
+			});
+		}
+		this.setState({ typeError: '' });
+		axios
+			.post('/api/dashboard/form', {
+				content: this.state.newQuestionText,
+				type: this.state.newQuestionType,
+			})
+			.then(({ data }) => {
+				if (!data.message) return this.setState({ error: data.message });
+				const newId = this.state.questions.length;
```

Comment:
this is bad. If two questions have the same length, they will have the same id.

use UUID v4 to generate a unique id.